### DISCUSSION TITLE
New to Raspberry Pi and Pool Equipment - Need some advice.

### DISCUSSION THREAD

**Discussion by spifftek70** [tagyoureit/nodejs-poolController#567] [My Pool]
I bought a house and now have both an in-ground pool and hot tub.  They were build in the late '70s and have NEVER been updated.  The Hot tub propane heater went out years ago, so the owners boarded over the hot tub.  When I moved in, I had a Pentair UltraTemp heat/cooler unit put in for the pool.  It has a Pool and Spa mode, so it can take the place a a propane heater (just not as fast).  I have to manually turn in/out valves to change flow to the UltraTemp between the pool and spa systems.  It's a pain.

Here's what I have:
<img width="1314" alt="Screen Shot 2022-08-10 at 11 38 36 AM" src="https://user-images.githubusercontent.com/11727190/183992348-1c1e440f-45eb-4894-a4f2-dee9a4a0d090.png">

Here is my planning for automation, but I need help.  
<img width="918" alt="Screen Shot 2022-08-10 at 11 36 05 AM" src="https://user-images.githubusercontent.com/11727190/183992590-2821c3ef-8ace-47cc-8990-6c70126a6b8e.png">
I'm not sure on the wiring diagram:

Suggestions / recommendations and explanations are appropriated.  Thanks in advance.

**Comment by tagyoureit** [comment]
Hi, and welcome!  I would definitely check out the [njsPC wiki](https://github.com/tagyoureit/nodejs-poolController/wiki#direct-pool-control-interfaces) which discusses a lot of this.  Likewise the [REM Wiki](https://github.com/rstrouse/relayEquipmentManager/wiki) has even more great nuggets.

A few things I can see and point out:
1. You don't need separate RS485 adapters.  One will suffice (although they do sometimes go out, so having a spare isn't a bad idea).
2. We love the [Sequent Microsystems cards](https://sequentmicrosystems.com/).  You'll see them referenced throughout the wikis.  They stack easily and work very well and we have full support for them.  It also significantly reduces the number of wires you need to hook up.
3.  Some of the more popular RS485 adapters are listed [here](https://github.com/tagyoureit/nodejs-poolController/wiki/RS-485-Adapter-Details#step-2) although most of them will work fine.
4.  For the Ultratemp, you can connect it via RS485, but there is no advantage to doing that vs. a fireman's switch from your relay.  
5.  For the Ultratemp, you will need a temp sensor feed into njsPC to let it know when to turn on the heater.  You cannot use the temp sensor in Ultratemp for that.
6. If you can get the Intellivalve instead of Intermatic, they are pretty good and don't need occasional adjustment like with the traditional valves.  Intellivalves don't yet (maybe ever) support RS485 but are otherwise solid valves.
7. You'll need a [24VAC power supply](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#24vac-power-supply) for the valves no matter which ones you get.

**Reply by spifftek70** [reply]
Excellent!  I'll check out the Wiki's this evening.  Thanks for the feedback too... Based on your recommendation, my diagram should look more like this (I need to figure out the exact wiring to the RasPi Stack):
<img width="1080" alt="Screen Shot 2022-08-10 at 4 32 40 PM" src="https://user-images.githubusercontent.com/11727190/184040860-e42636e4-801b-4495-aaa2-b1d67cfb0ae5.png">

**Reply by rstrouse** [reply]
There is an extensive read on the DIY Nixie Pool Controller.

**Reply by rock-crusher** [reply]
Hi spifftek70, you may want to consider the 8 relay hat as opposed to the 4. With the 4 you are already maxed out if using the fireman’s switch. I’m sure you will want to control your pool/spa lights and any lighting/pool features you may have now, or in the future.

**Reply by spifftek70** [reply]
Actually, I went with[ Eight RELAYS 8-Layer Stackable HAT](https://www.amazon.com/dp/B07KRKS67G?psc=1&ref=ppx_yo2ov_dt_b_product_details) for Raspberry Pi instead.

**Comment by spifftek70** [comment]
Okay, so I have my workroom covered in boxes, plastic, parts, tools..  I think I have the final hardware configuration set in this diagram which is very much taken from the Nixie Controller page instructions.  It's slightly different as it is tailored to my pool system, of course.  Do I have this wired correctly?

<img width="1395" alt="Screen Shot 2022-08-19 at 11 17 27 AM" src="https://user-images.githubusercontent.com/11727190/185682198-239d91ed-f720-422e-a5b8-93ce28fd98ea.png">

**Reply by rstrouse** [reply]
Not to be punny but that looks spiffy.  I assume you are installing a couple of 10k Temp sensors along with this though.  While your Ultratemp is a little bit smarter than a typical switched gas valve, it is still not smart enough to tell you what it thinks the temperature is.  This is really easy to do and very inexpensive.  Get a two pack and let yourself know how what the air temp is as well.

**Reply by spifftek70** [reply]
Please stop PUN-ishing me.  🤣 
You mean something like this?  Actually I'm installing four Temp sensors.  Pool, Spa, Air, and Solar.... even though I don't have a solar system, but apparently that is required by the UltraTemp system.

![IMG_5872](https://user-images.githubusercontent.com/11727190/185678708-c62a0903-96c8-48a9-86d1-afe988062f87.jpeg)

**Reply by rstrouse** [reply]
If you are running a shared body you will only want one water temp sensor.  If it is a dual body system then yes you want 2 water sensors.  The solar sensor is only because of idiosyncrasies with Pentair controllers when it comes to UltraTemp.  But you can install that sensor if you like.  

The primary reason for it is because the Pentair controllers can only use the solar sensor to check the efficiency rating for the heatpump but Nixie does not have this limitation.  It would make sense if you had both a heat pump and solar but beyond that it doesn't make any sense.  I think they opted to simplify the preference selection to only having to examine a specific sensor.

**Reply by spifftek70** [reply]
My pool setup is posted at the top of the page.  Separate pumps, and filters, but they share the UltraTemp through (currently manual) two values.  One value flowing out to the UltraTemp and one for the back from the UtlraTemp.  Is that semi-shared?

**Reply by rstrouse** [reply]
Oh that's right.  I would call that a dual body since both filter pumps can be running at the same time but it shares a heater.

**Reply by rock-crusher** [reply]
> Okay, so I have my workroom covered in boxes, plastic, parts, tools.. I think I have the final hardware configuration set in this diagram which is very much taken from the Nixie Controller page instructions. It's slightly different as it is tailored to my pool system, of course. Do I have this wired correctly?
> 
> <img alt="Screen Shot 2022-08-19 at 11 17 27 AM" width="1395" src="https://user-images.githubusercontent.com/11727190/185682198-239d91ed-f720-422e-a5b8-93ce28fd98ea.png">



Not to be picky, but I fixed it for you… lol 



<img width="833" alt="wiring change" src="https://user-images.githubusercontent.com/94593954/185689529-8ea5b1ab-700f-4b5a-a16f-b733f3153f6f.png">

**Reply by spifftek70** [reply]
LOL thanks

**Reply by rstrouse** [reply]
Yep that made it spiffier and so that it would actually work without voltage on the DIN rails.

**Reply by spifftek70** [reply]
yeah....  true.

**Reply by BigGass** [reply]
@spifftek70 What did you use to draw your wiring diagrams? I am starting to work on my own system and I want to layout my components and DIN rails. I've been struggling to find a way to draw them. Your drawings are so clear. I'm impressed. Thanks!

**Reply by rstrouse** [reply]
Some of those pics are from the wiki https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller and those were generated with Microsoft Visio.  @spifftek70 will have to comment on the others.

**Reply by spifftek70** [reply]
Thanks for the compliments @BigGass. 
I used [Miro](https://miro.com) to do that schematic.
You can also use [draw.io](https://draw.io) which is very similar.
Visio is also a good application, but it isn't free.

**Reply by BigGass** [reply]
Thank you both for your replies. I used to use Visio back when I was working (retired now) and its a really good program, but it is also really expensive. I will check our Miro and draw.io.
